### DISCUSSION TITLE
Notification from email doesn't work

### DISCUSSION THREAD

**Discussion by tipmg** [Unleash/unleash#5323] [Q&A] [answered]
Good day, Unleash team.

I deployed the application to k8s (5.6.5 version) with environment variables which include [Email Service](https://docs.getunleash.io/using-unleash/deploy/email-service).

The log unleash-server start:
`[2023-11-10T13:01:12.700] [INFO] services/email-service.ts - Initialized transport to mail.test.infra on port 25 with user: unleash@test.tech
[2023-11-10T13:01:12.833] [DEBUG] /middleware/pat-middleware.ts - Enabling PAT middleware
[2023-11-10T13:01:12.833] [DEBUG] /middleware/api-token.ts - Enabling api-token middleware
`
I didn't get any errors or warnings.

When I created a new user in unleash from UI interface with notification from email. I got an error and traceback from unleash-server. The user was created but notification about it was not delivered. Of course, I has already checked network connectivity and authorization to email server by command line.

The log unleash-server traceback:
`[2023-11-13T06:13:01.955] [DEBUG] controller/UserAdminController - I encountered an error that wasn't an instance of the UnleashError type. The original error was: { "input": "unleash.dev1.test.tech/new-user?token=token", "code": "ERR_INVALID_URL" }. It was mapped to { "id": "id", "name": "UnknownError", "message": "Invalid URL", "details": [ { "message": "Invalid URL", "description": "Invalid URL" } ] } [2023-11-13T06:13:01.955] [ERROR] controller/UserAdminController - Server failed executing request: { "input": "unleash.dev1.test.tech/new-user?token=token", "code": "ERR_INVALID_URL" } TypeError [ERR_INVALID_URL]: Invalid URL at new NodeError (node:internal/errors:405:5) at new URL (node:internal/url:637:13) at ResetTokenService.createResetUrl (/unleash/node_modules/unleash-server/dist/lib/services/reset-token-service.js:63:32) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async UserAdminController.createUser (/unleash/node_modules/unleash-server/dist/lib/routes/admin-api/user-admin.js:348:31) at async /unleash/node_modules/unleash-server/dist/lib/routes/controller.js:45:17 { input: 'unleash.dev1.amediateka.tech/new-user?token=token', code: 'ERR_INVALID_URL' }`

Could you provide me information about it?

**Comment by nunogois** [comment]
Hi @tipmg, thanks for reaching out! Looking at the error you shared, it seems the input is expected to be a valid URL, including a protocol:

![image](https://github.com/Unleash/unleash/assets/14320932/edd8bbbb-9398-42ee-b5f3-b2d8c484e4c3)

Can you please try again while including the protocol? In this case, it might look something like: `smtp://unleash.dev1.test.tech/new-user?token=token`

Let us know if this helps. Thank you!

**Reply by tipmg** [reply]
Hello, @nunogois 
I added variables for Email Service. The variable EMAIL_SECURE has value true for using SMTP. Also, I added protocol to the variable UNLEASH_URL
Unfortunately, the issue is actual from UI but the application provided URL with token for a created account. https://unleash.dev1.test.tech/new-user?token=token
I checked the log unleash-server:
```
[WARN] services/email-service.ts - Failed to send getting started email [Error: 589B0F49237F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:
] {
  library: 'SSL routines',
  reason: 'wrong version number',
  code: 'ESOCKET',
  command: 'CONN'
}

```

**Reply by chriswk** [reply]
If you're hosting your own SMTP server which uses a custom SSL certificate, you will need to tell node about the cert it should trust.
This is usually done by mounting the custom certificate in a volume, and then adding the environment variable `NODE_EXTRA_CA_CERTS` and setting it to the absolute path in the container where it can find the certificate (this includes the filename of the certificate, so if you mount it to `/var/certs` and your SMTP server cert has the name `secure-smtp.crt` the variable should be set to `/var/certs/secure-smtp.crt`). 

Though, from the error message, it seems like you're connecting to the PLAIN port of your SMTP server rather than the TLS port. You should double check that your connection details are correct.

**Comment by chriswk** [comment, marked as answer]
So, there are two error messages here, one is complaining about not being able to construct a valid URL, the other is complaining about SSL version numbers.

For the first one, ensure your UNLEASH_URL is a valid URL, that is, don't exclude the protocol prefix `http://` `https://` when writing it out. If you host it on a subpath, remember to include the entire URL to reach your server in the UNLEASH_URL environment variable, and set BASE_PATH_URI to the suffix after your server name. In the error log above
```
input: 'unleash.dev1.amediateka.tech/new-user?token=token'
```
UNLEASH_URL should be `https://unleash.dev1.amediateka.tech` and BASE_PATH_URI can be empty, since it seems like you use subdomains rather than path variables to separate your hosted services :+1: 

For the second error, I inadvertently posted it as a reply to Nuno's response. The `incorrect version number` from SSLroutines is usually due to trying to connect to the PLAIN port instead of the TLS port. Similar response would happen if you tried to communicate on port 80 (default plain http port) to an https server (usually hosted on 443). First thing to look at here would be to ensure that you communicate on the TLS port of your SMTP server (typically 587, not 25)

**Reply by tipmg** [reply]
Hello, @chriswk 
Yes, you sure right. The variable NODE_EXTRA_CA_CERTS resolved the issue with SSL. @nunogois wrote a right solution with first issue. Thanks for support 👍

**Reply by chriswk** [reply]
Thank you for confirming that it fixed the issue for you. Happy to help. Hope Unleash solves your feature toggle needs.
